status/views.py

```diff
@@ -0,0 +1,107 @@
+from elasticsearch import Elasticsearch, ConnectionError as ESConnectionError
+from psycopg2 import connect, OperationalError
+from redis import StrictRedis, ConnectionError as RedisConnectionError
+
+log = logging.getLogger(__name__)
+
+
+def get_pg_info():
+    """Check PostgreSQL connection."""
+    try:
+        conf = settings.DATABASES['default']
+    except (AttributeError, KeyError):
+        log.error("No PostgreSQL connection info found in settings.")
+        return {"status": "no config found"}
+    try:
+        start = datetime.now()
+        connect(
+            database=conf["NAME"], user=conf["USER"], host=conf["HOST"],
+            port=conf["PORT"], password=conf["PASSWORD"]
+        )
+        micro = (datetime.now() - start).microseconds
+    except (OperationalError, KeyError):
+        log.error("Invalid PostgreSQL connection info in settings: %s", conf)
+        return {"status": "down"}
+    return {"status": "up", "response_ms": micro}
+
+
+def get_redis_info():
+    """Check Redis connection."""
+    pat = re.compile(r'redis://([^:]+):(\d+)/(\d+)$')
+    try:
+        url = settings.BROKER_URL
+    except AttributeError:
+        log.error("No Redis connection info found in settings.")
+        return {"status": "no config found"}
+    try:
+        match = pat.match(url)
+    except TypeError:
+        match = None
+    if match is None:
+        log.error("Invalid Redis connection info found in settings: %s", url)
+        return {"status": "no config found"}
+    host, port, database = match.groups()
+    rdb = StrictRedis(host=host, port=port, db=database)
+    start = datetime.now()
+    try:
+        info = rdb.info()
+    except RedisConnectionError:
+        return {"status": "down"}
+    micro = (datetime.now() - start).microseconds
+    ret = {
+        "status": "up", "response_ms": micro,
+    }
+    fields = ("uptime_in_seconds", "used_memory", "used_memory_peak")
+    ret.update({x: info[x] for x in fields})
+    return ret
+
+
+def get_elasticsearch_info():
+    """Check Elasticsearch connection."""
+    pat = re.compile(r'([^:]+):(\d+)$')
+    try:
+        url = settings.HAYSTACK_CONNECTIONS["default"]["URL"]
+    except (AttributeError, KeyError):
+        log.error("No Elasticsearch connection info in settings.")
+        return {"status": "no config found"}
+    match = pat.match(url)
+    if match is None:
+        log.error("Invalid Elasticsearch connection info: %s.", url)
+        return {"status": "no config found"}
+    host, port = match.groups()
+    search = Elasticsearch(hosts=[{"host": host, "port": port}])
+    start = datetime.now()
+    try:
+        info = search.info()
+    except ESConnectionError:
+        return {"status": "down"}
+    micro = (datetime.now() - start).microseconds
+    return {
+        "status": "up", "response_ms": micro, "status_code": info["status"]
+    }
+
+
+def status(request):  # pylint: disable=unused-argument
+    """Status"""
+    info = {}
```

Comment:
So I was thinking, since connections are a limited resource, and these do "cost" us.  It would be nice to just do a simple query string type token that is defined in settings to protect this page (i.e. just return a 403 or 404).  Health checkers take a URL and all of the ones I've run into honor the `?=` querystring type params